todo_app/app.py

```diff
@@ -1,49 +1,127 @@
-from flask import Flask, render_template, request, redirect, url_for
+from flask import Flask, render_template, request, redirect, url_for, abort
+from flask_login import LoginManager, login_required
+import flask_login
+from flask_login.mixins import UserMixin
+from oauthlib.oauth2 import WebApplicationClient
+from oauthlib.oauth2.rfc6749.grant_types import client_credentials
 import Mongo_items as mongo
 import viewmodel as vm
+import os, requests, json
+from functools import wraps
 
 def create_app():
     app = Flask(__name__)
     app.config.from_object('flask_config.Config')
 
+    login_manager = LoginManager()
+
+    @login_manager.unauthorized_handler
+    def unauthenticated():
+        client = WebApplicationClient(os.getenv("CLIENT_ID"))
+        authredirect = client.prepare_request_uri("https://github.com/login/oauth/authorize", state="saxasdfaa")
+        return redirect(authredirect)
+
+    @login_manager.user_loader
+    def load_user(user_id):
+        user = User(user_id)
+        return user
+
+    login_manager.init_app(app)
+
+    class User(UserMixin):
+        def __init__(self, userid):
+            self.user_id = userid
```

Comment:
Note that you never use this `user_id` field so we could drop it entirely in favour of `id`